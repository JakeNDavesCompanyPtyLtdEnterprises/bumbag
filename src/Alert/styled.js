import styled from 'reakit/styled';
import { palette, theme } from 'styled-tools';
import { darken } from 'polished';
import { Box } from '../primitives';
import Label from '../Label';

export const AlertTitle = styled(Label)`
  & {
    ${theme('Alert.title.base')};
  }
`;

export default styled(Box)`
  background-color: ${props =>
    props.palette === 'default' ? palette('textTint')(props) : palette(`${props.palette}Tint`)(props)};
  border-left: 4px solid
    ${props => (props.palette === 'default' ? palette('text')(props) : palette(props.palette)(props))};
  border-radius: 4px;
  color: ${props =>
    props.palette === 'default' ? darken(0.25, palette('text')(props)) : darken(0.25, palette(props.palette)(props))};
  padding: 1rem;

  & {
    ${theme('Alert.base')};
  }
`;
