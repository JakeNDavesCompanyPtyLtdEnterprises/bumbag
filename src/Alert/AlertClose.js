// @flow
import React from 'react';
import { AlertClose as _AlertClose } from './styled';
import Icon from '../Icon';
import VisuallyHidden from '../VisuallyHidden';

type Props = {
  isAbsolute?: boolean,
  onClickClose?: Function
};

const AlertClose = ({ onClickClose, ...props }: Props) => (
  <_AlertClose kind="link" onClick={onClickClose} {...props}>
    <VisuallyHidden>Close</VisuallyHidden>
    <Icon aria-hidden="true" icon="cross" />
  </_AlertClose>
);

AlertClose.defaultProps = {
  isAbsolute: false,
  onClickClose: undefined
};

export default AlertClose;
