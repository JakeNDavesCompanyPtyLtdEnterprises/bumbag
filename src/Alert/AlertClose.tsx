import * as React from 'react';

import { ButtonProps } from '../Button/Button';
import Icon from '../Icon';
// @ts-ignore
import VisuallyHidden from '../VisuallyHidden';
import { Omit } from '../types';
import { AlertClose as _AlertClose } from './styled';

export interface LocalAlertCloseProps {
  isAbsolute?: boolean;
  onClickClose?: ButtonProps['onClick'];
}
export type AlertCloseProps = Omit<ButtonProps, 'children'> & LocalAlertCloseProps;

export const AlertClose: React.FunctionComponent<LocalAlertCloseProps> = ({ onClickClose, ...props }) => (
  <_AlertClose kind="link" onClick={onClickClose} {...props}>
    <VisuallyHidden>Close</VisuallyHidden>
    <Icon a11yHidden icon="times" />
  </_AlertClose>
);

AlertClose.defaultProps = {
  isAbsolute: false,
  onClickClose: undefined
};

const C: React.FunctionComponent<AlertCloseProps> = AlertClose;
export default AlertClose;
