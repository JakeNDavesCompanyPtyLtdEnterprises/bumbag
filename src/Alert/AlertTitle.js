// @flow
import React, { type Node } from 'react';
import { AlertTitle as _AlertTitle } from './styled';

type Props = {
  children: Node
};

const AlertTitle = ({ children, ...props }: Props) => (
  <_AlertTitle as="h6" isSemiBold {...props}>
    {children}
  </_AlertTitle>
);

export default AlertTitle;
