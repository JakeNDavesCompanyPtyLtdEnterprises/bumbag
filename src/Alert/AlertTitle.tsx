import * as React from 'react';
import { AlertTitle as _AlertTitle } from './styled';

export interface AlertTitleProps {
  children: React.ReactNode;
}

const AlertTitle = ({ children, ...props }: AlertTitleProps) => (
  <_AlertTitle use="h6" isSemiBold {...props}>
    {children}
  </_AlertTitle>
);

export default AlertTitle;
