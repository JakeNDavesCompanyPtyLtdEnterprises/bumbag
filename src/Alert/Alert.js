// @flow
import React, { type Node } from 'react';

import { Box, Flex } from '../primitives';
import styled, { css } from '../styled';
import Icon from '../Icon';
import Text from '../Text';
import _Alert from './styled';
import AlertTitle from './AlertTitle';

const ContentWrapper = styled(Box)`
  padding: 1rem;

  ${props =>
    props.hasIcon &&
    css`
      padding-left: 0;
    `};
`;
const IconWrapper = styled(Flex)`
  padding: 0 1rem;
`;

type Props = {
  className?: string,
  children?: Node,
  hasIcon?: boolean,
  hasTint?: boolean,
  title?: string,
  type?: 'info' | 'success' | 'danger' | 'warning'
};

const Alert = ({ className, children, hasIcon, title, type, ...props }: Props) => (
  <_Alert role="alert" className={className} type={type} {...props}>
    <Flex alignItems="center">
      {hasIcon && (
        <IconWrapper>
          <Icon color={type} icon={type} size={children ? 'medium' : undefined} />
        </IconWrapper>
      )}
      <ContentWrapper hasIcon={hasIcon}>
        {title && <AlertTitle>{title}</AlertTitle>}
        {typeof children === 'string' ? <Text>{children}</Text> : children}
      </ContentWrapper>
    </Flex>
  </_Alert>
);

Alert.defaultProps = {
  className: undefined,
  children: undefined,
  hasIcon: false,
  hasTint: false,
  title: undefined,
  type: 'info'
};

export default Alert;
