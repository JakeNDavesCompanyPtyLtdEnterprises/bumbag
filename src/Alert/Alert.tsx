import * as React from 'react';

import _Alert from './styled';
import AlertTitle from './AlertTitle';
// @ts-ignore
import Text from '../Text';

export interface AlertProps {
  className?: string;
  children?: React.ReactNode;
  hasTint?: boolean;
  palette?: string;
  title?: string;
}

const defaultProps: Partial<AlertProps> = {
  className: undefined,
  children: undefined,
  hasTint: false,
  palette: 'default',
  title: undefined
};

export const Alert: React.SFC<AlertProps> = ({ className, children, palette, title, ...props }) => (
  <_Alert role="alert" className={className} palette={palette} {...props}>
    {title && <AlertTitle>{title}</AlertTitle>}
    {typeof children === 'string' ? <Text>{children}</Text> : children}
  </_Alert>
);

Alert.defaultProps = defaultProps;

export default Alert;
