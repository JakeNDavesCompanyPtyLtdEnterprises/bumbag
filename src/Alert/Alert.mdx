---
name: Alert
route: /components/alert
menu: Components
---

import { Playground, PropsTable } from 'docz';
import Alert from '../Alert';

# Alerts

## Import

`import { Alert } from 'fannypack'`

## Basic Usage

<Playground>
  <Alert>Anim pariatur cliche reprehenderit, enim eiusmod high life accusamus terry richardson ad squid. Nihil anim keffiyeh helvetica.</Alert>
</Playground>

## Titles

You can set a title with the `title` prop.

<Playground>
  <Alert title="Alert Title">Anim pariatur cliche reprehenderit, enim eiusmod high life accusamus terry richardson ad squid. Nihil anim keffiyeh helvetica.</Alert>
  <Alert title="Alert Title" marginTop="xsmall" />
</Playground>

## Types

An alert can take a type of `"info"`, `"success"`, `"warning"` or `"danger"`.

<Playground>
  <Alert title="Alert Title" type="info" marginTop="xsmall">Anim pariatur cliche reprehenderit, enim eiusmod high life accusamus terry richardson ad squid. Nihil anim keffiyeh helvetica.</Alert>
  <Alert title="Alert Title" type="success" marginTop="xsmall">Anim pariatur cliche reprehenderit, enim eiusmod high life accusamus terry richardson ad squid. Nihil anim keffiyeh helvetica.</Alert>
  <Alert title="Alert Title" type="warning" marginTop="xsmall">Anim pariatur cliche reprehenderit, enim eiusmod high life accusamus terry richardson ad squid. Nihil anim keffiyeh helvetica.</Alert>
  <Alert title="Alert Title" type="danger" marginTop="xsmall">Anim pariatur cliche reprehenderit, enim eiusmod high life accusamus terry richardson ad squid. Nihil anim keffiyeh helvetica.</Alert>
</Playground>

## Icons

You can add a icon to your alert by using the `hasIcon` prop. The icon will default to the status' icon. If you want to change this icon, then configure the `iconNames` attribute in the [Icon theme](/components/icon#theming).

<Playground>
  <Alert title="Alert Title" hasIcon type="info" marginTop="xsmall">Anim pariatur cliche reprehenderit, enim eiusmod high life accusamus terry richardson ad squid. Nihil anim keffiyeh helvetica.</Alert>
  <Alert title="Alert Title" hasIcon type="success" marginTop="xsmall">Anim pariatur cliche reprehenderit, enim eiusmod high life accusamus terry richardson ad squid. Nihil anim keffiyeh helvetica.</Alert>
  <Alert title="Alert Title" hasIcon type="warning" marginTop="xsmall">Anim pariatur cliche reprehenderit, enim eiusmod high life accusamus terry richardson ad squid. Nihil anim keffiyeh helvetica.</Alert>
  <Alert title="Alert Title" hasIcon type="danger" marginTop="xsmall">Anim pariatur cliche reprehenderit, enim eiusmod high life accusamus terry richardson ad squid. Nihil anim keffiyeh helvetica.</Alert>
</Playground>

## Inline alerts

To inline an alert, set the `display` prop to `inline-block`.

<Playground>
  <Alert display="inline-block">This is an inline alert.</Alert>
</Playground>

## Tints

<Playground>
  <Alert title="Alert Title" hasTint type="info" marginTop="xsmall">Anim pariatur cliche reprehenderit, enim eiusmod high life accusamus terry richardson ad squid. Nihil anim keffiyeh helvetica.</Alert>
  <Alert title="Alert Title" hasTint type="success" marginTop="xsmall">Anim pariatur cliche reprehenderit, enim eiusmod high life accusamus terry richardson ad squid. Nihil anim keffiyeh helvetica.</Alert>
  <Alert title="Alert Title" hasTint type="warning" marginTop="xsmall">Anim pariatur cliche reprehenderit, enim eiusmod high life accusamus terry richardson ad squid. Nihil anim keffiyeh helvetica.</Alert>
  <Alert title="Alert Title" hasTint type="danger" marginTop="xsmall">Anim pariatur cliche reprehenderit, enim eiusmod high life accusamus terry richardson ad squid. Nihil anim keffiyeh helvetica.</Alert>
</Playground>

## Props

<PropsTable of={Alert} />

## Theming

### Schema

```jsx
{
  base: string | Object,
  title: {
    base: string | Object
  }
}
```
