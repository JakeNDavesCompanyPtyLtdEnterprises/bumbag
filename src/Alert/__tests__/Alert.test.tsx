import * as React from 'react';
import render from '../../_utils/tests/render';
// @ts-ignore
import Alert from '../Alert.tsx';

it('renders correctly', () => {
  const { container } = render(<Alert>Alert</Alert>);
  expect(container.firstChild).toMatchSnapshot();
});

it('renders correctly with a custom element', () => {
  const { container } = render(
    <Alert>
      <label>Alert</label>
    </Alert>
  );
  expect(container.firstChild).toMatchSnapshot();
});

it('renders correctly when a palette prop is set', () => {
  const { container } = render(<Alert palette="primary">Alert</Alert>);
  expect(container.firstChild).toMatchSnapshot();
});

it('renders correctly when a title prop is set', () => {
  const { container } = render(<Alert title="Title">Alert</Alert>);
  expect(container.firstChild).toMatchSnapshot();
});
