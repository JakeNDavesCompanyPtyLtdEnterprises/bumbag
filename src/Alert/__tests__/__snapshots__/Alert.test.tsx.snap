// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
.c1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: unset;
  color: #fcfcfc;
}

.c3 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: unset;
  color: inherit;
}

.c0 {
  background-color: white;
  border: 1px solid #e6e6e6;
  border-left: 4px solid #435a6f;
  border-radius: 4px;
  color: #435a6f;
  padding: 1rem;
}

abbr.c2 {
  border-bottom: 1px dotted;
  cursor: help;
  -webkit-text-decoration: none;
  text-decoration: none;
}

i.c2 {
  font-style: italic;
}

kbd.c2 {
  background: #454d5d;
  border-radius: 0.1rem;
  color: #fff;
  fill: #fff;
  padding: 0.1rem 0.2rem;
}

mark.c2 {
  background: #ffe9b3;
  border-radius: 0.1rem;
  padding: 0.1rem 0.2rem;
}

strong.c2 {
  font-weight: bold;
}

sub.c2,
sup.c2 {
  font-size: 0.8em;
  vertical-align: baseline;
}

ins.c2,
u.c2 {
  -webkit-text-decoration-skip: ink edges;
  text-decoration-skip: ink edges;
}

i.c2,
cite.c2,
var.c2,
em.c2 {
  font-style: italic;
}

kbd.c2,
pre.c2,
samp.c2 {
  font-family: 'SF Mono','Segoe UI Mono','Roboto Mono',Menlo,Courier,monospace;
}

.c2 .c4 {
  top: 0.1em;
}

<div
  class="c0 c1"
  role="alert"
>
  <span
    class="c2 c3"
  >
    Alert
  </span>
</div>
`;

exports[`renders correctly when a palette prop is set 1`] = `
.c3 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: unset;
  color: inherit;
}

.c1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: unset;
  color: #3926a5;
}

.c0 {
  background-color: white;
  border: 1px solid #e6e6e6;
  border-left: 4px solid #3926a5;
  border-radius: 4px;
  color: #3926a5;
  padding: 1rem;
}

abbr.c2 {
  border-bottom: 1px dotted;
  cursor: help;
  -webkit-text-decoration: none;
  text-decoration: none;
}

i.c2 {
  font-style: italic;
}

kbd.c2 {
  background: #454d5d;
  border-radius: 0.1rem;
  color: #fff;
  fill: #fff;
  padding: 0.1rem 0.2rem;
}

mark.c2 {
  background: #ffe9b3;
  border-radius: 0.1rem;
  padding: 0.1rem 0.2rem;
}

strong.c2 {
  font-weight: bold;
}

sub.c2,
sup.c2 {
  font-size: 0.8em;
  vertical-align: baseline;
}

ins.c2,
u.c2 {
  -webkit-text-decoration-skip: ink edges;
  text-decoration-skip: ink edges;
}

i.c2,
cite.c2,
var.c2,
em.c2 {
  font-style: italic;
}

kbd.c2,
pre.c2,
samp.c2 {
  font-family: 'SF Mono','Segoe UI Mono','Roboto Mono',Menlo,Courier,monospace;
}

.c2 .c4 {
  top: 0.1em;
}

<div
  class="c0 c1"
  role="alert"
>
  <span
    class="c2 c3"
  >
    Alert
  </span>
</div>
`;

exports[`renders correctly when a title prop is set 1`] = `
.c1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: unset;
  color: #fcfcfc;
}

.c4 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: unset;
  color: inherit;
}

.c3 {
  font-weight: 700;
  line-height: 1.2;
  margin-bottom: 1rem;
}

.heading.c3 + .sub-heading {
  margin-top: -1rem;
}

.c3:last-child {
  margin-bottom: 0;
}

h1.c3 {
  font-size: 3rem;
}

h2.c3 {
  font-size: 2.5rem;
}

h3.c3 {
  font-size: 2rem;
}

h4.c3 {
  font-size: 1.5rem;
}

h5.c3 {
  font-size: 1.25rem;
}

h6.c3 {
  font-size: 1rem;
}

.c3 .c6 {
  top: 0.1em;
}

.c2 {
  font-weight: 600;
  margin-bottom: 0.5rem;
}

.c0 {
  background-color: white;
  border: 1px solid #e6e6e6;
  border-left: 4px solid #435a6f;
  border-radius: 4px;
  color: #435a6f;
  padding: 1rem;
}

abbr.c5 {
  border-bottom: 1px dotted;
  cursor: help;
  -webkit-text-decoration: none;
  text-decoration: none;
}

i.c5 {
  font-style: italic;
}

kbd.c5 {
  background: #454d5d;
  border-radius: 0.1rem;
  color: #fff;
  fill: #fff;
  padding: 0.1rem 0.2rem;
}

mark.c5 {
  background: #ffe9b3;
  border-radius: 0.1rem;
  padding: 0.1rem 0.2rem;
}

strong.c5 {
  font-weight: bold;
}

sub.c5,
sup.c5 {
  font-size: 0.8em;
  vertical-align: baseline;
}

ins.c5,
u.c5 {
  -webkit-text-decoration-skip: ink edges;
  text-decoration-skip: ink edges;
}

i.c5,
cite.c5,
var.c5,
em.c5 {
  font-style: italic;
}

kbd.c5,
pre.c5,
samp.c5 {
  font-family: 'SF Mono','Segoe UI Mono','Roboto Mono',Menlo,Courier,monospace;
}

.c5 .c6 {
  top: 0.1em;
}

@media (max-width:480px) {
  h1.c3 {
    font-size: 2rem;
  }
}

@media (max-width:480px) {
  h2.c3 {
    font-size: 1.8rem;
  }
}

@media (max-width:480px) {
  h3.c3 {
    font-size: 1.6rem;
  }
}

@media (max-width:480px) {
  h4.c3 {
    font-size: 1.4rem;
  }
}

@media (max-width:480px) {
  h5.c3 {
    font-size: 1.2rem;
  }
}

<div
  class="c0 c1"
  role="alert"
>
  <h6
    class="c2 heading c3 c4"
  >
    Title
  </h6>
  <span
    class="c5 c4"
  >
    Alert
  </span>
</div>
`;

exports[`renders correctly with a custom element 1`] = `
.c1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: unset;
  color: #fcfcfc;
}

.c0 {
  background-color: white;
  border: 1px solid #e6e6e6;
  border-left: 4px solid #435a6f;
  border-radius: 4px;
  color: #435a6f;
  padding: 1rem;
}

<div
  class="c0 c1"
  role="alert"
>
  <label>
    Alert
  </label>
</div>
`;
